Sources/EmojiText/NSEmojiTextView.swift

```diff
@@ -0,0 +1,201 @@
+
+            // Load emojis. Will set placeholders for lazy emojis
+            renderedEmojis = renderedEmojis.merging(loader.loadEmojis(emojis)) { current, new in
+                if current.hasSameSource(as: new) {
+                    if !new.isPlaceholder || current.isPlaceholder {
+                        return new
+                    } else {
+                        return current
+                    }
+                } else {
+                    return new
+                }
+            }
+            await self.render(renderedEmojis)
+            // Load emojis. Will set placeholders for lazy emojis
+            renderedEmojis = renderedEmojis.merging(await loader.loadLazyEmojis(emojis)) { current, new in
+                if current.hasSameSource(as: new) {
+                    if !new.isPlaceholder || current.isPlaceholder {
+                        return new
+                    } else {
+                        return current
+                    }
+                } else {
+                    return new
+                }
+            }
+            await self.render(renderedEmojis)
+        }
+    }
+
+    private func render(_ renderedEmojis: [String: LoadedEmoji]) {
+        let string: NSAttributedString = renderer.render(
+            string: raw,
+            emojis: renderedEmojis,
+            size: nil
+        )
+        let result = NSMutableAttributedString(attributedString: string)
+        result.enumerateAttribute(.link) { value, range, stop in
+            guard value is URL else { return }
+            result.addAttribute(.foregroundColor, value: NSColor.controlAccentColor, range: range)
+        }
+        guard let textStorage else { return }
+        textStorage.setAttributedString(result)
+    }
+
+    func makeLoader() -> EmojiLoader {
+        EmojiLoader(placeholder: placeholder, font: font ?? NSFont.preferredFont(forTextStyle: .body)) { parameter in
+            parameter
+            // .overrideSize(size)
+            // .overrideBaselineOffset(baselineOffset)
+                .displayScale(window?.screen?.backingScaleFactor)
+        }
+        .emojiProvider(syncEmojiProvider: syncEmojiProvider, asyncEmojiProvider: asyncEmojiProvider)
+    }
+
+    var placeholder: any CustomEmoji {
+        if let image = NSImage(systemName: "square.dashed") {
+            return LocalEmoji(shortcode: "placeholder", image: image, color: .placeholderEmoji, renderingMode: .template)
+        } else {
+            return SFSymbolEmoji(shortcode: "placeholder", symbolRenderingMode: .monochrome, renderingMode: .template)
+        }
+    }
+}
+
+#if DEBUG
+import SwiftUI
+
+private struct NSPreview: NSViewRepresentable {
+    func makeNSView(context: Context) -> NSView {
+        NSEmojiTextView(markdown: "**Hello** :iphone: _and_ :a:", emojis: .emojis)
+    }
+
+    func updateNSView(_ nsView: NSView, context: Context) {
+    }
+}
+
+#Preview {
+    NSPreview()
+        .padding()
+}
+#endif
```

Comment:
I don't know why you wrapped it in `NSViewRepresentable` when `NSEmojiTextView` is an AppKit class to be used mainly in AppKit context.

```suggestion
#Preview {
    NSEmojiTextView(markdown: "**Hello** :iphone: _and_ :a:", emojis: .emojis)
}
```

For some reasons, it gave me this error:

> 'Preview(_:traits:body:)' is only available in macOS 14.0 or newer